packages/react-ui/src/widgets/FeatureSelectionWidgetUI/FeatureSelectionWidgetUI.js

```diff
@@ -51,25 +53,68 @@ function FeatureSelectionWidgetUI({
   onDeleteGeometry,
   tooltipPlacement = 'bottom',
   size = 'medium',
-  chipLabel
+  chipLabel,
+  locale
 }) {
+  const intl = useImperativeIntl(locale);
+
+  const selectionModeWithLocales = useMemo(() => {
+    return selectionModes.map((mode) => ({
+      ...mode,
+      label: intl.formatMessage({
+        id: `c4r.widgets.featureSelection.${
+          mode.id.charAt(0).toLowerCase() + mode.id.slice(1)
```

Comment:
please create util function for this logic

Reply:
Done